src/engine/guidance/segment_list.cc

```diff
@@ -0,0 +1,231 @@
+#include "segment_list.hpp"
+
+#include <cstddef>
+
+#include <boost/assert.hpp>
+
+namespace osrm
+{
+namespace engine
+{
+namespace route
```

Comment:
the namespace should correspond with the filesystem structure. Like `engine/guidance/ ` -> `osrm::engine::guidance`.